### DISCUSSION TITLE
Separate include files for Debug and Release

### DISCUSSION THREAD

**Discussion by human-0** [microsoft/vcpkg#18703] [Ideas]
Currently, the header files are the same in Debug and Release mode. This is a problem in libraries such as GLAD, which have different include files. During build time, different headers are used (in `buildtrees\...\x64-windows-dbg\include` and `buildtrees\...\x64-windows-rel\include`, but only the release headers are available in `x64-windows\include`. This makes it impossible to access the debug only features of the library, such as `glad_set_pre_callback` and `glad_set_post_callback`.

To fix this, different include files are needed for Debug and Release. A current workaround is to copy the contents of the debug header to the release header and conditionally select the implementation using the preprocessor. Obviously, this is not ideal, especially in other libraries where there may be many header files.

**Comment by JackBoosY** [comment]
It is not necessary to store multiple identical files in installed, we can create _debug/include_ link to _include_ to solve the single build type issue like this.
